### PR TITLE
ci: upload workflow run data to GCS daily

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds a workflow to upload the previous day's workflow 
run information to a GCS bucket

### CODE REVIEW

File: .github/workflows/ci-data.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: generate data
+        id: generate_data
+        run: |
+          set -euo pipefail
+
+          yesterday="${{ steps.set_date.outputs.yesterday }}"
+
+          [ -n "$yesterday" ] || exit 1
+
+          gh api --paginate "repos/ibis-project/ibis/actions/runs?per_page=100&created=${yesterday}" --jq '.workflow_runs[]' > workflows.json
```
Comment (kszucs) [inline .github/workflows/ci-data.yml:28]: Does it use the `GITHUB_TOKEN` provided by github actions? In that case the rate limit is just 1000, so sleeping one second won't help much.

File: .github/workflows/ci-data.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      - run: gcloud info
+
+      - name: copy to gcs
+        run: |
+          set -euo pipefail
+
+          yesterday="${{ steps.set_date.outputs.yesterday }}"
+
+          gsutil cp -Z workflows.json "gs://ibis-workflow-data/${yesterday}/workflows.json"
```
Comment (jreback) [inline .github/workflows/ci-data.yml:52]: not a big deal but do we need daily? wouldn't it be better to have this run when the deps of this change?

Comment (cpcloud) [reply .github/workflows/ci-data.yml:52]: Dependencies are unrelated here. I want to store all actions runs data.

File: .github/workflows/ci-data.yml (hunk 1 above)
Comment (cpcloud) [reply .github/workflows/ci-data.yml:32]: Fixed.

**Comment by cpcloud** [comment]
@jreback can you approve?

**Review by jreback — APPROVED** [review]

**Pull request merged**